### PR TITLE
Improve core.after in a simple way

### PR DESCRIPTION
**Pull request by DS-Minetest**
There were already PRs that wanted to make `core.after` faster, but it was argued against them because of their complexity.
In this PR there's
- no complicated data structure used.
- `table.remove` replaced with swapping the last element with the element to remove (O(1) instead of [O(n)](https://github.com/minetest/minetest/blob/b298b0339c79db7f5b3873e73ff9ea0130f05a8a/lib/lua/src/ltablib.c#L125-L128)).
- a better best case for checking the job expires (by saving the lowest next time). (This is the same as in  #6037.)
- No worse runtime at adding new jobs.

(n is the number of elements in the jobs table, k is the number of expired jobs)
globalstep runtimes:

case | old | new
--- | --- | ---
best case (k = 0) | O(n) | O(1)
average case | O(k*n) | O(n)
worst case (k = n) | O(k*n) = O(n^2) | O(n)

new after job runtimes:

 old | new
 --- | ---
 O(1) | O(1)

Is a change like this wanted or is this already too complicated and too hard to maintain?

### CODE REVIEW

**Comment by SmallJoker** [comment]
This is a simple but also effective performance improvement, thanks for this contribution! It needs some quick tests, but the code already looks good in style and logic.

**Comment by DS-Minetest** [comment]
After simple testing I haven't found mistakes. However, I don't know how to test this reliably.

File: builtin/common/after.lua [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-	if #jobs < 1 then
+	if time < time_next then
 		return
 	end
 
+	time_next = math.huge
+
 	-- Iterate backwards so that we miss any new timers added by
 	-- a timer callback, and so that we don't skip the next timer
 	-- in the list if we remove one.
```
Comment (pgimeno) [inline builtin/common/after.lua]: This comment is now obsolete. The second reason is no longer relevant. The first reason is still valid, though; it implies that if a callback uses `minetest.after(0, ...)`, the new timer won't execute until the next globalstep.

File: builtin/common/after.lua [hunk 2]
```diff
@@ -16,18 +19,23 @@ core.register_globalstep(function(dtime)
 		if time >= job.expire then
 			core.set_last_run_mod(job.mod_origin)
 			job.func(unpack(job.arg))
-			table.remove(jobs, i)
+			jobs[i] = jobs[#jobs]
+			jobs[#jobs] = nil
```
Comment (pgimeno) [inline builtin/common/after.lua]: Use a variable for `#jobs` to avoid calculating it twice.

File: builtin/common/after.lua [hunk 3]
```diff
@@ -21,8 +21,8 @@ core.register_globalstep(function(dtime)
 			job.func(unpack(job.arg))
 			jobs[i] = jobs[#jobs]
 			jobs[#jobs] = nil
-		else
-			time_next = math.min(time_next, job.expire)
+		elseif job.expire < time_next then
+			time_next = job.expire
```
Comment (pgimeno) [inline builtin/common/after.lua]: `math.min` is faster than the extra branch. Please revert this change.

Comment (pgimeno) [reply builtin/common/after.lua]: Sorry, never mind. This code won't be compiled, and in interpreted mode the `if` version is faster.

File: builtin/common/after.lua (hunk 2 above)
Comment (DS-Minetest) [reply builtin/common/after.lua]: Done.

File: builtin/common/after.lua (hunk 1 above)
Comment (DS-Minetest) [reply builtin/common/after.lua]: Done.

**Comment by pgimeno** [comment]
LGTM FWIW :)
